USAGE.rst

```diff
@@ -0,0 +1,137 @@
+
+Let's start simple. Give ``kafkacat`` a broker and a topic, and you'll see its contents:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t mysql_users
+    % Auto-selecting Consumer mode (use -P or -C to override)
+    {"uid":1,"name":"Cliff","locale":"en_US","address_city":"St Louis","elite":"P"}
+    {"uid":2,"name":"Nick","locale":"en_US","address_city":"Palo Alto","elite":"G"}
+    [...]
+
+
+Note: ``kafkacat`` doesn't currently support Avro.
+
+``kafkacat`` defaults to being a consumer, but you can also make this explicit with the ``-C`` flag. You can also specify how many messages you want to see with the lowercase ``-c`` and a number. For example:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t mysql_users -C -c1
+    {"uid":1,"name":"Cliff","locale":"en_US","address_city":"St Louis","elite":"P"}
+
+Seeing the value of the message is useful, but Kafka messages also have a key, which ``kafkacat`` can display with the ``-K`` argument also supplying the delimiter:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t mysql_users -C -c1 -K\t
+    1   {"uid":1,"name":"Cliff","locale":"en_US","address_city":"St Louis","elite":"P"}
+
+With ``kafkacat`` we can really dig into a topic's messages. The ``-f`` flag takes arguments specifying both the format of the output, as well as the fields to include. Here's a simple example of pretty-printing the key/value pairs for each message:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t mysql_users -C -c1 -f 'Key: %k\nValue: %s\n'
+    Key: 1
+    Value: {"uid":1,"name":"Cliff","locale":"en_US","address_city":"St Louis","elite":"P"}
+
+(Note that we replaced the ``-K:`` because the key parameter is specified in the ``-f`` format string now)
+
+A more advanced use of ``-f`` would be to show even more metadata - offsets, timestamps, and even data lengths:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t mysql_users -C -c2 -f '\nKey (%K bytes): %k\t\nValue (%S bytes): %s\nTimestamp: %T\tPartition: %p\tOffset: %o\n--\n'
+
+    Key (1 bytes): 1
+    Value (79 bytes): {"uid":1,"name":"Cliff","locale":"en_US","address_city":"St Louis","elite":"P"}
+    Timestamp: 1520618381093        Partition: 0    Offset: 0
+    --
+
+    Key (1 bytes): 2
+    Value (79 bytes): {"uid":2,"name":"Nick","locale":"en_US","address_city":"Palo Alto","elite":"G"}
+    Timestamp: 1520618381093        Partition: 0    Offset: 1
+    --
+
+``kafkacat`` the Producer
+-------------------------
+
+Sending data to a topic is really simple with ``kafkacat`` too. Run it with the ``-P`` command and enter the data you want, and then press Ctrl-D to finish:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t new_topic -P
+    test
+
+Replay it (replace ``-P`` with ``-C``) to verify:
+
+.. code:: shell
+
+    $ kafkacat -b localhost:9092 -t new_topic -C
+    test
+
+As well as ``stdin`` you can add data from a file (``-l``), and using the ``-T`` flag to also echo the input to ``stdout``:
```

Comment:
suggest `You can send data to ``stdin`` by adding data from a file (``-l``), and using the ``-T`` flag to also echo the input to ``stdout```